src/lib/mock-observations/helpers/rand-int-generator.ts

```diff
@@ -46,6 +46,14 @@ export const RandIntGenerator = (
         })
       );
     }
+
+    if (config.min >= config.max) {
+      throw new InputValidationError(
+        errorBuilder({
+          message: `Min value should not be greater than max value of ${validatedName}`,
+        })
+      );
+    }
     return {min: config.min, max: config.max};
```

Comment:
I know that it's Paz's code, however, min and max can be used with restructuring and Object.prototype.hasOwnProperty can be replaced by 'min' in config, 'max' in config